hr_expense_invoice/models/account_move.py

```diff
@@ -1,11 +1,26 @@
 # Copyright 2019 Ecosoft <saranl@ecosoft.co.th>
+# Copyright 2021 Tecnativa - Víctor Martínez
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
-from odoo import models
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
 
 
 class AccountMove(models.Model):
     _inherit = "account.move"
 
+    expense_ids = fields.One2many(
+        comodel_name="hr.expense", inverse_name="invoice_id", string="Expenses"
+    )
+
+    @api.constrains("amount_total")
+    def _check_expense_ids(self):
+        for move in self:
+            if move.expense_ids:
+                if sum(move.expense_ids.mapped("total_amount")) != move.amount_total:
+                    raise UserError(
```

Comment:
On constraints, it should be `ValidationError`:
```suggestion
                    raise ValidationError(
```